File: conformance/tests/if-else/if-else.runar.py
from runar import SmartContract, Bigint, public, assert_

class IfElse(SmartContract):
    limit: Bigint

    def __init__(self, limit: Bigint):
        super().__init__(limit)
        self.limit = limit

    @public
    def check(self, value: Bigint, mode: bool):
        result: Bigint = 0
        if mode:
            result = value + self.limit
        else:
            result = value - self.limit
        assert_(result > 0)

File: conformance/tests/if-else/if-else.runar.rb
require 'runar'

class IfElse < Runar::SmartContract
  prop :limit, Bigint

  def initialize(limit)
    super(limit)
    @limit = limit
  end

  runar_public value: Bigint, mode: Boolean
  def check(value, mode)
    result = 0
    if mode
      result = value + @limit
    else
      result = value - @limit
    end
    assert result > 0
  end
end
